File: tchat-api/Tchat.Api.Domain/Message.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace Tchat.Api.Domain
{
    public class Message
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [NotNull]
        [Required]
        public string Content { get; set; }
        [NotNull]
        [Required]
        public DateTime SendDateTime { get; set; }
        [ForeignKey(nameof(User))]
        public string UserId { get; set; }
        [NotNull]
        [Required]
        public string TchatIp { get; set; }
        public User User { get; set; }
    }
}

File: tchat-web/src/modules/model/Message.ts
export default class Message {
  private _id: string
  private _content: string
  private _sentDateTime: Date
  private _userId: string
  private _username: string

  public get userId(): string {
    return this._userId
  }

  public get content(): string {
    return this._content
  }

  public get id(): string {
    return this._id
  }

  public get sendDateTime(): Date {
    return this._sentDateTime
  }

  public get username(): string {
    return this._username
  }

  constructor(
    id: string,
    content: string,
    sentDateTime: Date,
    userId: string = '',
    username: string = ''
  ) {
    this._id = id
    this._content = content
    this._sentDateTime = sentDateTime
    this._userId = userId
    this._username = username
  }
}
